### DISCUSSION TITLE
Future Game Ideas

### DISCUSSION THREAD

**Discussion by shoc71** [shoc71/dont-stop-now#75] [Ideas]
Found the perfect spot to put for level themes and future functions;

- [x] night mode (either inverted colors and/or certain black boxes are not always colliding)
- [ ] no checkboxes mode 
- [ ] mostly red
- [ ] coins?
- [ ] double-jump levels (maybe)
- [ ] fog
- [ ] wind

See if you can more ideas below.

**Comment by shoc71** [comment]


**Reply by slysnivy** [reply]
I also added these to the main post

**Reply by shoc71** [reply]
I don't know why this is separate from the actual list but I think we can either ignore or remove this now.

**Reply by slysnivy** [reply]
Yea, feel free to delete

**Comment by slysnivy** [comment]
Can you elaborate on what checkboxes and mostly red are? @shoc71

**Reply by shoc71** [reply]
I have no idea what I was thinking that long ago, I can only assume it's like a key mode, where you have to collect all the keys before you can pass the level. @slysnivy

**Reply by slysnivy** [reply]
What's mostly red?

**Reply by shoc71** [reply]
The levels are at least 70% red meaning that 70% of the screen is red like the player is inside the mouth of a volcano

**Comment by slysnivy** [comment]
We should also consider the effects that double jump has on the existing levels. Would they make the levels unexpectedly easier? Would that be a good or bad thing.

**Reply by shoc71** [reply]
If we enable the double-jump function for all the current levels that have been already created that would destroy the game entirely. I did mention earlier during the inception of this game because I had some level ideas for this. For this point now, I was thinking after a certain level count, we enable double-jump for only certain levels.

**Reply by slysnivy** [reply]
How about we turn it into a power up? Just so the player can actively know/choose to have double jump. Then, we can choose to include it in the levels we want. So it'll be a good indicator to players if they have double jump for that level and we can implement this in LevelScene (collision based power up) and SquareMe (allow jump_ability variable to be True twice, likely using a counter).

**Reply by shoc71** [reply]
Could you elaborate more on this? It sounds like the player can pick and choose whichever levels they want to use the double-jump function on. As mentioned before, some levels are enjoyable with single-jump and double-jump would totally ruin the experience.

**Reply by slysnivy** [reply]
It will be a power up **_inside_** the level. So it'll be up to the player, **in** the level, to pick up the double jump power up. We can design the levels to punish the player for either not picking up or deciding to pick up the power up.

It'll be implemented as another collision object **inside** the level.

In that way, we can design the level to include the double jump or not, freely **(not up to the player to decide)**

I hope that helps! 👍

**Reply by shoc71** [reply]
Interesting (yes the explanation above helps)... Let's have this as a secret-level function for now and other levels for it to be unlocked as.

**Reply by slysnivy** [reply]
Are you saying that it would be included **both** as a power up and as a level unlock? 
What I was thinking was that it would just be a power up.
Could you explain more about the level unlock part?

**Reply by shoc71** [reply]
Yes, like many of the game function ideas mentioned above - it will get its own series of levels dedicated to that double-jump function. Once the main branch is complete, the player will have access (code for double-jump) in level-zero, enter it, and play that series of levels completing it one at a time. As you mentioned, the player will start the game normally and then collide with the superpower and then they can double-jump as freely as they want to, in order to, complete the level.

**Reply by slysnivy** [reply]
Ok I understand. It's a power up exclusive to those levels. 👍

**Comment by slysnivy** [comment]
We need a teleporter

**Reply by shoc71** [reply]
hmmm..... maybe next game or something like that

**Comment by shoc71** [comment]
Achievement Ideas for one of the filler classes to fill up.
![kogideas](https://github.com/shoc71/Keep-On-Going/assets/129419169/1d5215fc-64a5-4dff-9c2b-fd030c9e51e8)

**Reply by shoc71** [reply]
@slysnivy

**Comment by shoc71** [comment]
Teleportation Ideas;
- [Jumphase](https://www.coolmathgames.com/0-jumphase) 
- [Yin and Yang ](https://www.coolmathgames.com/0-yin-and-yang)
- [Mirror Wizard](https://www.coolmathgames.com/0-mirror-wizard)

**Reply by slysnivy** [reply]
I played through all 3 of these games and I think we can do something similar to mirror wizard, or telegraphing where the player will teleport if they can. The problem is how we're going to include it story-wise (since most of these don't have an actual story upfront).

**Reply by slysnivy** [reply]
I think the best place to include elaborate powerups like these is in the post-game due to 2 reasons:
1. This gives time for the player to master the base mechanics and gives them the opportunity to expand on them.
2. Since post-game (after the final boss) the level zero gate gives access/freedom to the player to explore different realms, which means different mechanics etc.
